src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets

```diff
@@ -2341,10 +2341,21 @@ because xbuild doesn't support framework reference assemblies.
     ResolvedAssemblies="@(_ResolvedAssemblies)"
     ResolvedUserAssemblies="@(_ResolvedUserAssemblies)"
     MainAssembly="$(MonoAndroidLinkerInputDir)$(TargetFileName)"
-    OutputDirectory="$(IntermediateOutputPath)android\src\mono" 
+    OutputDirectory="$(IntermediateOutputPath)android\src\mono"
+    EnvironmentOutputDirectory="$(IntermediateOutputPath)android\src\mono\android\app"
```

Comment:
Because conceptually it is *application* environment, so it belongs in the app package and also, `src/mono/android/app` is created *anyway* so we're reusing an existing location anyway.